File: presentation/src/main/java/com/yalin/style/view/fragment/BaseFragment.java
package com.yalin.style.view.fragment;

import android.app.Fragment;
import android.widget.Toast;

import com.yalin.style.injection.HasComponent;

/**
 * @author jinyalin
 * @since 2017/4/20.
 */

public abstract class BaseFragment extends Fragment {
    /**
     * Shows a {@link android.widget.Toast} message.
     *
     * @param message An string representing a message to be shown.
     */
    protected void showToastMessage(String message) {
        Toast.makeText(getActivity(), message, Toast.LENGTH_SHORT).show();
    }

    /**
     * Gets a component for dependency injection by its type.
     */
    @SuppressWarnings("unchecked")
    protected <C> C getComponent(Class<C> componentType) {
        return componentType.cast(((HasComponent<C>) getActivity()).getComponent());
    }
}

File: presentation/src/main/java/com/yalin/style/view/fragment/BaseFragment.kt
package com.yalin.style.view.fragment

import android.app.Fragment
import android.widget.Toast

import com.yalin.style.injection.HasComponent

/**
 * @author jinyalin
 * *
 * @since 2017/4/20.
 */

abstract class BaseFragment : Fragment() {
    /**
     * Shows a [android.widget.Toast] message.

     * @param message An string representing a message to be shown.
     */
    protected fun showToastMessage(message: String) {
        Toast.makeText(activity, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * Gets a component for dependency injection by its type.
     */
    protected fun <C> getComponent(componentType: Class<C>): C {
        @Suppress("UNCHECKED_CAST")
        return componentType.cast((activity as HasComponent<C>).component)
    }
}
